### DISCUSSION TITLE
Dynamically creating a QueryConstraint

### DISCUSSION THREAD

**Discussion by bvanderw** [netreconlab/Parse-Swift#165] [Q&A]
I would like to be able to dynamically create queries (e.g. from user input). In the ObjC code, you could use an NSPredicate, but it is not clear how it can be done in Parse-Swift. It seems the easiest way would be to use this initializer on QueryConstraint, but it is not public.

`
    init(key: String, value: Codable? = nil, comparator: Comparator? = nil, isNull: Bool = false)
`

Can someone point me in the right direction?

**Comment by cbaker6** [comment]
You can use mutations and build a constant array similar to the examples in the playgrounds https://github.com/netreconlab/Parse-Swift/blob/702ca34b47a9512d39cc24d4b5b540391bcafea5/ParseSwift.playground/Pages/7%20-%20GeoPoint.xcplaygroundpage/Contents.swift#L115-L140
